beacon_node/network/src/sync/network_context.rs

```diff
@@ -211,14 +216,23 @@ pub struct SyncNetworkContext<T: BeaconChainTypes> {
     /// A mapping of active DataColumnsByRange requests
     data_columns_by_range_requests:
         ActiveRequests<DataColumnsByRangeRequestId, DataColumnsByRangeRequestItems<T::EthSpec>>,
-
+    data_columns_by_root_range_requests:
+        ActiveRequests<DataColumnsByRootRequestId, DataColumnsByRootRangeRequestItems<T::EthSpec>>,
     /// Mapping of active custody column requests for a block root
     custody_by_root_requests: FnvHashMap<CustodyRequester, ActiveCustodyRequest<T>>,
 
     /// BlocksByRange requests paired with other ByRange requests for data components
     components_by_range_requests:
         FnvHashMap<ComponentsByRangeRequestId, RangeBlockComponentsRequest<T::EthSpec>>,
 
+    // todo(pawan): make this a bounded queue, make the types better, add better docs
+    // A hashmap with the key being the parent request and the value being the data column by root
+    // requests that we have to retry because of one of the following reasons:
+    // 1. The root requests couldn't be made after the parent blocks request because there were no
+    // column peers available
+    // 2. The root request errored (either peer sent an RPC error or an empty response)
+    requests_to_retry: HashMap<ComponentsByRangeRequestId, DataColumnsByRootBatchBlockRequest>,
```

Comment:
This name is generic however it's specifically for columns by root in range sync.
Could name this to something like `range_column_by_root_requests_to_retry` or `pending_column_by_root_range_requests`?

Reply:
I like pending_column_by_root_range_requests. Renamed